syntaxes/GDShader.tmLanguage.json

```diff
@@ -314,51 +314,51 @@
         },
         "literalFloat": {
             "name": "constant.numeric.float.gdshader",
-            "match": "\b(?:\\d+[eE][-+]?\\d+|(?:\\d*[.]\\d+|\\d+[.])(?:[eE][-+]?\\d+)?)[fF]?"
+            "match": "\\b(?:\\d+[eE][-+]?\\d+|(?:\\d*[.]\\d+|\\d+[.])(?:[eE][-+]?\\d+)?)[fF]?"
         },
         "literalInt": {
             "name": "constant.numeric.integer.gdshader",
-            "match": "\b(?:0[xX][0-9A-Fa-f]+|\\d+[uU]?)\b"
+            "match": "\\b(?:0[xX][0-9A-Fa-f]+|\\d+[uU]?)\\b"
         },
         "literalBool": {
             "name": "constant.language.boolean.gdshader",
-            "match": "\b(?:false|true)\b"
+            "match": "\\b(?:false|true)\\b"
         },
         "identifierType": {
             "name": "entity.name.type.gdshader",
-            "match": "\b[a-zA-Z_]\\w*(?=(?:\\s*\\[\\s*\\w*\\s*\\])?\\s+[a-zA-Z_]\\w*\b)"
+            "match": "\\b[a-zA-Z_]\\w*(?=(?:\\s*\\[\\s*\\w*\\s*\\])?\\s+[a-zA-Z_]\\w*\\b)"
         },
         "constructor": {
             "name": "entity.name.type.constructor.gdshader",
-            "match": "\b[a-zA-Z_]\\w*(?=\\s*\\[\\s*\\w*\\s*\\]\\s*[(])|\b[A-Z]\\w*(?=\\s*[(])"
+            "match": "\\b[a-zA-Z_]\\w*(?=\\s*\\[\\s*\\w*\\s*\\]\\s*[(])|\\b[A-Z]\\w*(?=\\s*[(])"
         },
         "processorFunction": {
             "name": "support.function.gdshader",
-            "match": "\b(?:vertex|fragment|light|start|process|sky|fog)(?=(?:\\s|/\\*(?:\\*(?!/)|[^*])*\\*/)*[(])"
+            "match": "\\b(?:vertex|fragment|light|start|process|sky|fog)(?=(?:\\s|/\\*(?:\\*(?!/)|[^*])*\\*/)*[(])"
         },
         "identifierFunction": {
             "name": "entity.name.function.gdshader",
-            "match": "\b[a-zA-Z_]\\w*(?=(?:\\s|/\\*(?:\\*(?!/)|[^*])*\\*/)*[(])"
+            "match": "\\b[a-zA-Z_]\\w*(?=(?:\\s|/\\*(?:\\*(?!/)|[^*])*\\*/)*[(])"
         },
         "swizzling": {
             "name": "variable.other.property.gdshader",
-            "match": "(?<=[.]\\s*)(?:[xyzw]{2,4}|[rgba]{2,4}|[stpq]{2,4})\b"
+            "match": "(?:[.]\\s*)(?:[xyzw]{2,4}|[rgba]{2,4}|[stpq]{2,4})\\b"
         },
         "identifierField": {
             "name": "entity.name.variable.field.gdshader",
-            "match": "(?<=[.]\\s*)[a-zA-Z_]\\w*\b"
+            "match": "(?:[.]\\s*)[a-zA-Z_]\\w*\\b"
         },
```

Comment:
Just a minor comment on these. If lookbehind for the dot is replaced with making it part of the captured regex, then the editor will likely include the dot too as part of the field/property coloring. Which is fine if that's what is intended.
But in case this is undesired, you can put `[.]` in a capturing group and use something like:
```json
"captures": {
  "1": {
    "name": "punctuation.accessor.gdshader"
  }
},
```

Reply:
I updated both rules, and added the example file you linked me to in a previous discussion.

I think my original version was actually broken, so thanks for the feedback. The new version actually works, as well as correctly tagging the period. I'd greatly appreciate it if you could proofread me again!